lib/structuraid_core/engineering/optimization/rebar_constant_long.rb

```diff
@@ -0,0 +1,126 @@
+require 'interactor'
+
+module StructuraidCore
+  module Engineering
+    module Optimization
+      # Runs optimization of the reinforcement, it asumes that the rebar length is constant, so if any shosen rebar has the same length as any other, the target is just to minimize the total mass by changing the amount of rebars
+      class RebarConstantLong
+        include Interactor
+
+        OPTIONAL_REBAR_NUMBERS = [3, 4, 5, 6, 7].freeze
+        MINIMUM_REBAR_AMOUNT = 2
+        UNSUCCESSFUL_RESULT_MESSAGE = 'required area exceeds maximum spacing'.freeze
+        SUCCESSFUL_RESULT_MESSAGE = 'done'.freeze
+
+        # @param required_reinforcement_area [Float] The reinforcement area required by the design
+        # @param maximum_rebar_spacing [Float] The maximum spacing between rebars
+        # @param space_to_cover [Float] The space to cover with the reinforcement, the rebars will be placed with a spacing which must be less than maximum_rebar_spacing
+        def initialize(required_reinforcement_area, maximum_rebar_spacing, space_to_cover)
+          @required_reinforcement_area = required_reinforcement_area
+          @maximum_rebar_spacing = maximum_rebar_spacing
+          @space_to_cover = space_to_cover
+        end
+
+        def run
+          @steps_log = []
+          run_optimization
+          return unsuccessful_result if current_rebar.nil?
+
+          successful_result
+        end
+
+        def log
+          steps_log
+        end
```

Comment:
I kinda like the pattern you are trying to implement here. I think there are mixed concerns, though. Maybe we can later try to refactor this stuff to have in a module all the logic regarding logging the results of each iteration, and keep here the iteration logic which is the actual optimization logic. That way the logging module could be reused on all the other optimization processes we build, and the optimization logic here would remain clear after cleaning out all the logging things

It would be awesome, but definitely that sounds like a next iteration for this class and the next optimization process we build

Reply:
Ufff right, specially because there are more optimizators to build